app/main.py

```diff
@@ -39,51 +48,257 @@ def execute(self):
 
             self.execute_as_loop(30)  # 30-second interval.
         """
+        self.load_shelve()
 
     def shutdown(self):
         """Run when your NApp is unloaded.
 
         If you have some cleanup procedure, insert it here.
         """
 
+    @listen_to("kytos/topology.unloaded")
+    def unload_topology(self):  # pylint: disable=W0613
+        """Function meant for validation, to make sure that the shelve
+        has been loaded before all the other functions that use it begins to
+        call it."""
+        self.shelve_loaded = False  # pylint: disable=W0201
+
+    def test_kytos_topology(self):
+        """ Test if the Topology napp has loaded """
+        try:
+            _ = self.get_kytos_topology()
+            return True
+        except Exception as err:  # pylint: disable=W0703
+            log.debug(err)
+            return False
+
+    @staticmethod
+    def get_kytos_topology():
+        """retrieve topology from API"""
+        kytos_topology = requests.get(
+                settings.KYTOS_TOPOLOGY, timeout=10).json()
+        return kytos_topology["topology"]
+
+    def validate_sdx_topology(self):
+        """ return 200 if validated topology following the SDX data model"""
+        try:
+            response = requests.post(
+                    settings.SDX_TOPOLOGY_VALIDATE,
+                    json=self.sdx_topology,
+                    timeout=10)
+        except ValueError as exception:
+            log.info("validate topology result %s %s", exception, 401)
+            raise HTTPException(
+                    401,
+                    detail=f"Path is not valid: {exception}"
+                ) from exception
+        return {"result": response.json(), "status_code": response.status_code}
+
+    def convert_topology(self, event_type=0, event_timestamp=None):
+        """Function that will take care of update the shelve containing
+        the version control that will be updated every time a change is
+        detected in kytos topology, and return a new sdx topology"""
+        try:
+            # open the topology shelve
+            with shelve.open("topology_shelve") as open_shelve:
+                version = open_shelve['version']
+                self.dict_shelve = dict(open_shelve)  # pylint: disable=W0201
+                open_shelve.close()
```

Comment:
To ensure that changes to shelves are properly saved, it's important to use either close or sync. This way, when kytos is running and the code reads the shelves, it will have access to the most up-to-date information.

Reply:
This `open_shelve.close()` close here is redundant, since you've opened a context manager [it'll close it](https://github.com/python/cpython/blob/main/Lib/shelve.py#L137-L138). 

Please sync with @italovalcy's suggestion regarding either only storing in memory or when persistence is needed reaching out to MongoDB. Assuming that this NApp will also run in the official docker container, notice that the shelve file might not be in a docker volume depending how it's mounted so this result in other problems considering how it's deployed at AmLight.